dist/pythonlibs/riotctrl_shell/cord_ep.py

```diff
@@ -0,0 +1,145 @@
+        'coap://[fe80::48d5:9eff:fe98:6b74]:5683'
+        >>> res['epname']
+        'RIOT-760D2323760D2323'
+        >>> res['ltime']
+        60
+        >>> res['regif']
+        '/resourcedirectory'
+        >>> res = parser.parse("CoAP RD connection status:\\n"
+        ...             "RD address:"
+        ...             " coaps://[fe80::48d5:9eff:fe98:6b74%iface0]:5684\\n"
+        ...             "   ep name: RIOT-760D2323760D2323\\n"
+        ...             "  lifetime: 60s\\n"
+        ...             "    reg if: /resourcedirectory\\n"
+        ...             "  location: /reg/1/")
+        >>> res['rdaddr']
+        'coaps://[fe80::48d5:9eff:fe98:6b74%iface0]:5684'
+        """
+        res = {}
+        for line in cmd_output.splitlines():
+            for key, comp in self.comps.items():
+                m = comp.search(line)
+                if m is not None:
+                    try:
+                        res[key] = int(m.group(key))
+                    except ValueError:
+                        res[key] = m.group(key)
+        return res if bool(res) else None
+
+
+class CordEpDiscoverParser(ShellInteractionParser):
+
+    def __init__(self):
+        self.comp = re.compile(r"the registration interface is"
+                               r"\s+'(?P<regif>\S+)'$")
+
+    def parse(self, cmd_output):
+        """
+        Parses output of CordEp::cord_ep_discover()
+
+        >>> parser = CordEpDiscoverParser()
+        >>> parser.parse(
+        ...     "the registration interface is '/resourcedirectory'")
+        '/resourcedirectory'
+        """
+        for line in cmd_output.splitlines():
+            m = self.comp.search(line)
+            if m is not None:
+                return m.group("regif")
+        return None
+
+
+# ==== ShellInteractions ====
+
+
+class CordEp(ShellInteraction):
+    REGISTER = "register"
+    DISCOVER = "discover"
+    UPDATE = "update"
+    REMOVE = "remove"
+    INFO = "info"
+
+    @staticmethod
+    def _make_addr(addr, iface=None, port=None):
+        if iface is not None:
+            addr += "%{}".format(iface)
+        addr = f"[{addr}]"
+        if port is not None:
+            addr += ":{}".format(port)
+        return addr
+
+    @ShellInteraction.check_term
+    def cord_ep_cmd(self, cmd, args=None, timeout=-1, async_=False):
+        cmd_str = "cord_ep {cmd}".format(cmd=cmd)
+        if args is not None:
+            cmd_str += " {args}".format(args=" ".join(str(a) for a in args))
+        res = self.cmd(cmd_str, timeout=timeout, async_=async_)
+        if ("error:" in res) or ("usage:" in res):
+            raise RuntimeError(res)
+        return res
+
+    def cord_ep_info(self, timeout=-1, async_=False):
+        return self.cord_ep_cmd(self.INFO, None, timeout, async_)
+
+    def cord_ep_register(self, addr, iface=None, port=None, regif=None,
+                         timeout=-1, async_=False):
+        args = [self._make_addr(addr, iface, port)]
```

Comment:
Mhh.. why did you split the URI up now? This makes things more complicated than they need to (just one thing coming to mind: how would I now register via `coaps` `coap+uart` or over TCP once the register command supports all that?

Reply:
Oh, just realized this was the case before, you just added `iface`. Anyway, an `uri` parameter would be prefered here, I think.

Reply:
I forgot to consider coap over other channel. Removed the split.